### ISSUE TITLE
Removing old blocks

### ISSUE THREAD

**Issue by HardeepAsrani** [Codeinwp/otter-blocks#16]
I was thinking about the future of the plugin and had a quick discussion about it Ionut. And after a lot of feedback from a lot of people, and some of my own ideas, I wanted to propose that we remove a lot of old blocks from Otter.

Our main focus should be:

- Advanced Columns (which includes Button Group and Advanced Heading as a companion)
- Posts Grid - All the people who use this plugin use it for Posts Grid, at least the ones I've talked to.

Apart from this, we should keep:

- Font Awesome Icons - Because we use it in Templates.
- About Author and Sharing Icons - Same reason as above.
- Google Map - It's also useful for templates and can see people using it.

All other blocks are bloat:

- Notice & Click to Tweet are not that amazing anyway, so we can remove them.
- I absolutely hate Accordion block, and it's the worst.
- Pie chart can be removed, and we can recommend people to use Visualizer instead.
- Plugin Card is fine, and I think we can release it as a separate plugin.

We can release the plugin and remove them from Insertor but people who are already using it in their posts will still find it working.

- For Notice, Click to Tweet and Accordion we can make a separate plugin and release it using my personal account or JFT (anything but company account) just for people who want to keep it. They won't be maintained much and if not many people use it, we can later remove it from plugin repo.

- Pie chart will be replaced by Visualizer, and Plugin card can be released separately (maybe we can work to make it better too).

---

Apart from this, the next version will remove the following blocks for new users:

- Pricing, Services and Testimonials.

They will still appear in the spots they're being used, but you won't be able to add new ones. They will become a part of the Template Library. Users will see an option to add their Singular block, but that will just be a block which will append its markup without any custom settings, just so people can easily add new Pricing/ Service and Testimonial element.

---

So basically, Otter will have Template/Columns and companion blocks that are needed for it. What do you think? @ineagu @selul

**Comment by selul** [comment]
@HardeepAsrani I get your point, but I'm not sure that making a separate plugin worth to try. For charts, I agree that we should keep just in Visualizer.

But for Notice, Click to Tweet and Accordion making a separate plugin doesn't make too much sense for me, either we remove them completely or we keep them further. Another solution is to keep them only in OrbitFox. 

I'm not sure I understand the change for the last 3 blocks, Pricing, services, and testimonials. What exactly will change for them? What will differ from current flow?

Another thing here, don't know if is possible to know what blocks a particular site is using? If yes, how about how many times was that block used on that site? I'm thinking we can leverage this in the future to know which blocks are used more, to have some data backed decisions.

**Comment by HardeepAsrani** [comment]
> But for Notice, Click to Tweet and Accordion making a separate plugin doesn't make too much sense for me, either we remove them completely or we keep them further.

I agree. That's why I propose we release that plugin from my account just for the sake of the people who want to keep using it. This way the company won't have to maintain it.

> Another solution is to keep them only in OrbitFox.

That will make things little complicated to maintain. :)

> I'm not sure I understand the change for the last 3 blocks, Pricing, services, and testimonials. What exactly will change for them? What will differ from current flow?

It's little hard to explain, so let me try. We will remove those three blocks from the plugin for adding. If you've added them to your post, they will still work. If you haven't added them, you won't see an option to add it.

Apart from it, we will add Structural blocks (it's not a Gutenberg term, something that I call them). Instead of having the old Pricing Table with three Pricing Elements, they will just consist of Pricing Elements. So people can easily use them to make custom templates.

![screenshot-proud-walrus w5 poopy life-2019 01 13-02-24-27](https://user-images.githubusercontent.com/2649903/51078417-a9440680-16da-11e9-9ae6-66251a4630bc.png)

They will work as a companion to our columns, without having any custom functionality, instead of using other blocks inside.

> Another thing here, don't know if is possible to know what blocks a particular site is using? If yes, how about how many times was that block used on that site? I'm thinking we can leverage this in the future to know which blocks are used more, to have some data backed decisions.

We will have to find a way to do it, not sure how to do it but we can find a way to get the user data in future, as an opt-in.

@selul

**Comment by selul** [comment]
@HardeepAsrani i get your point regarding the last 3 blocks, makes sense now. 

Regarding of the blocks, you think we should remove if we keep them how this will affect otter ? Will reduce performance, will increase the assets size, or how ? 
Also, I'm not sure if we keep just in obfx, why is hard to maintain? Isn't the same case if we keep them in a separate plugin?

**Comment by HardeepAsrani** [comment]
> you think we should remove if we keep them how this will affect otter ?

For the most blocks, I guess 20kbs of size in the build.

For that Chart block, it will reduce our build size by more than 1MB. :)

> Also, I'm not sure if we keep just in obfx, why is hard to maintain? Isn't the same case if we keep them in a separate plugin?

If we keep it in a separate non-ThemeIsle plugin, then we don't have to maintain it after one release. We can pretty much give it as an option for people who want to use it, or to continue using it until they find an alternative. But I guess we can just remove them.

Right now they will be disabled, but not removed so people can remove them and find alternatives for these small things. If we remove them straight, their editor will break and they will hate us. :)
